client/blocks/login/social.jsx

```diff
@@ -34,25 +33,27 @@ class SocialLoginForm extends Component {
 			return;
 		}
 
-		this.props.loginSocialUser( 'google', response.Zi.id_token ).then( () => {
+		let creatingAccountNotice = null;
+
+		this.props.loginSocialUser( 'google', response.Zi.id_token, () => {
+			const { notice } = this.props.infoNotice( this.props.translate( 'Creating your account' ) );
+			creatingAccountNotice = notice;
+			return true;
```

Comment:
Passing a predicate in the `shouldCreateAccount` parameter that always returns `true` makes me think this will always create an account. It also feels a little strange to mix promises and callbacks like this.

Could we just use promises instead, but move the `usersSocialNew` to an action? This seems easier to understand to me:

```js
this.props.loginSocialUser( 'google', response.Zi.id_token )
  .then( result => {
    // set success state
  } )
  .catch( error => {
    if ( error.code === 'unknown_user' ) {
      // create an account
    } else {
      // error notice
    }
  } );
```